lib/apis/v2/ippool.go

```diff
@@ -42,7 +42,7 @@ type IPPoolSpec struct {
 	IPIPMode IPIPMode `json:"ipipMode,omitempty" validate:"omitempty,ipipmode"`
 	// When nat-outgoing is true, packets sent from Calico networked containers in
 	// this pool to destinations outside of this pool will be masqueraded.
-	NATOutgoing bool `json:"natOutgoing,omitempty"`
+	NATOutgoing bool `json:"natOutgoing,omitempty" confignamev1:"nat-outgoing"`
```

Comment:
not sure if this is needed for IPPool fields. @robbrockbank?

Reply:
The confignamev1 annotation is only used for the config resources where we used to have a separate entry for each config key and now we have a single resource *and* have changed the names of some of the keys.

For IP pools we just convert the IPPool struct to the v1 struct as before.  The old YAML name should not matter.

Reply:
@bcreane :  is this trying to fix an issue with IPPools?

Reply:
@robbrockbank - nah. I noticed the field name changed with v2 and that other changed field names were annotated similarly. This fixes nothing, so we can discard the PR if you don't need to track `nat-outing` -> `natOutgoing`.

Reply:
Yeah I think we can close this PR